model_analyzer/config/generate/perf_analyzer_config_generator.py

```diff
@@ -179,13 +182,18 @@ def _create_non_concurrency_perf_config_params(self):
         return perf_config_params
 
     def _step(self):
+        self._step_concurrency()
+
         if self._done_walking_concurrencies():
             self._step_config()
-        else:
-            self._step_concurrency()
+
+            if not self._done_walking_configs():
```

Comment:
I'm clearly missing something with these 'if not' cases

Reply:
Same as above.